sdk/src/main/java/com/horizen/secret/SchnorrSecretSerializer.java

```diff
@@ -21,11 +24,19 @@ public static SchnorrSecretSerializer getSerializer() {
 
     @Override
     public void serialize(SchnorrSecret secret, Writer writer) {
-        writer.putBytes(secret.bytes());
+        writer.putInt(secret.secretBytes.length);
```

Comment:
For schnorr, private and public keys have some constant length that can be retrieved from cryptoprovider or sc-cryptolib.
So we can put bytes only, without length. Same as for other secret types.